deltachat-ffi/src/lib.rs

```diff
@@ -1192,17 +1191,19 @@ pub unsafe extern "C" fn dc_archive_chat(
         return;
     }
     let ffi_context = &*context;
-    let archive = if archive == 0 {
-        false
-    } else if archive == 1 {
-        true
-    } else {
-        return;
+    let archive_state = match archive {
+        0 => ArchiveState::Normal,
+        1 => ArchiveState::Archived,
+        2 => ArchiveState::Pinned,
+        _ => {
+            eprintln!("ignoring careless call to dc_archive_chat(): unknown archived state");
+            return;
```

Comment:
You have `ffi_context` by now, so no need to use the last-resort stderr printing.  You should call `ffi_context.warning()` here instead.